// babel-plugin-transform-react-handled-props/lib/util/Store.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Store = function Store() {
  var _this = this;

  _classCallCheck(this, Store);

  this.entries = {};
  this.hasImport = false;

  this.addProps = function (name, newProps) {
    _this.entries[name].props = _lodash2.default.union(_this.entries[name].props, newProps);
  };

  this.createEntry = function (name, path) {
    _this.entries[name] = { path: path, props: [] };
  };

  this.getEntries = function () {
    return _lodash2.default.map(_this.entries, function (_ref, identifier) {
      var path = _ref.path,
          props = _ref.props;

      return { identifier: identifier, path: path, props: props };
    });
  };

  this.hasEntry = function (name) {
    return _lodash2.default.has(_this.entries, name);
  };
};

exports.default = Store;
